stdlib/public/core/Collection.swift

```diff
@@ -260,9 +260,10 @@ extension CollectionType {
 
   /// Returns the first element of `self`, or `nil` if `self` is empty.
   ///
-  /// - Complexity: O(1)
+  /// - Complexity: O(1) in most cases, possibly O(N) for lazy collections.
   public var first: Generator.Element? {
-    return isEmpty ? nil : self[startIndex]
+    var gen = generate()
+    return gen.next()
```

Comment:
I'm not proposing that isEmpty be changed to use a generator. This is just the `first` property which, by definition, has to access the first element anyway.